### PR TITLE
Add guess methods with default mime

### PR DESCRIPTION
**Pull request by DenisKolodin**
PR to solve #11

### CODE REVIEW

**Comment by cybergeek94** [comment]
In truth, I would prefer to change most of these methods to return `Option<Mime>` instead and have `guess_mime_type` just do `.unwrap_or_else(octet_stream)`. That'll require a major version bump, though.

**Comment by DenisKolodin** [comment]
Your approach is better. Updated.
I've not squashed commits yet, but ready to do it if no changes you want.
We can keep it till new major version or add this functions with other names to prevent breaking change.

File: src/lib.rs [hunk 1]
```diff
@@ -18,27 +18,25 @@ mod mime_types;
...
 /// If `path` has no extension, or its extension has no known MIME type mapping,
-/// then the MIME type is assumed to be `application/octet-stream`.
+/// then None is returned.
 ///
 /// ##Note
 /// **Guess** is the operative word here, as there are no guarantees that the contents of the file
 /// that `path` points to match the MIME type associated with the path's extension.
 ///
 /// Take care when processing files with assumptions based on the return value of this function.
-pub fn guess_mime_type<P: AsRef<Path>>(path: P) -> Mime {
```
Comment (cybergeek94) [inline src/lib.rs]: This should probably be renamed to `mime_type_from_ext` or `guess_mime_type_opt` and then `guess_mime_type` should remain as returning `Mime` for convenience:

```rust
pub fn guess_mime_type<P: AsRef<Path>>(path: P) -> Mime { 
    mime_type_from_ext(path).unwrap_or_else(octet_stream)
}
```

File: src/lib.rs [hunk 2]
```diff
@@ -18,27 +18,25 @@ mod mime_types;
...
+pub fn guess_mime_type<P: AsRef<Path>>(path: P) -> Option<Mime> {
```
Comment (cybergeek94) [inline src/lib.rs]: This should probably be renamed to `mime_type_from_ext` or `guess_mime_type_opt` and then `guess_mime_type` should remain as returning `Mime` for convenience:

```rust
pub fn guess_mime_type<P: AsRef<Path>>(path: P) -> Mime { 
    mime_type_from_ext(path).unwrap_or_else(octet_stream)
}
```

**Comment by DenisKolodin** [comment]
I've added methods with `_opt` suffix. This names looks closer to legacy version.
Also I've squashed commits.

**Comment by cybergeek94** [comment]
Excellent! These changes shouldn't break existing usage of the API. Well done.

**Pull request merged**